python/spacewalk/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -1270,6 +1270,10 @@ def set_download_parameters(self, params, relative_path, target_file=None, check
         params["minrate"] = self.minrate
         params['proxies'] = get_proxies(self.proxy_url, self.proxy_user, self.proxy_pass)
 
+        if len(self.repo.urls) == 1:
+            # when we have only 1 url copy it to have a 2nd try
```

Comment:
Small suggestion:

```suggestion
            # When the repository has only 1 possible URL, we duplicate it and add it as possible next mirror to try.
            # This will cause reposync does a 2nd try in case of possible spurious errors while downloading files from this repository.
```